### PR TITLE
ci: Add rule for Firedancer review on native programs

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The Firedancer team maintains a line-for-line reimplementation of the native programs, and until native programs are moved to BPF, those implementations must exactly match their Agave counterparts. In the past, the Firedancer team has requested to be included in reviews that touch code in the native programs.

#### Summary of Changes

To make it harder to forget, add an automated message to any PRs that touch native programs. I've omitted test crates and the ZK Token Proof program, since I believe that won't be included in Firedancer.

Fixes #

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -183,6 +183,24 @@ pull_request_rules:
...
           Thank you for keeping the RPC clients in sync with the server API @{{author}}.
+  - name: Reminder to add Firedancer team to changes in `programs/`
+    conditions:
+      - or:
+        - files~=^programs/address-lookup-table/src/.*\.rs$
+        - files~=^programs/bpf_loader/src/.*\.rs$
+        - files~=^programs/compute_budget/src/.*\.rs$
+        - files~=^programs/config/src/.*\.rs$
+        - files~=^programs/loader-v4/src/.*\.rs$
+        - files~=^programs/stake/src/.*\.rs$
```
Comment (CriesofCarrots) [inline .mergify.yml:194]: What about the stake and vote files in `sdk/program`?

Comment (joncinque) [reply .mergify.yml:194]: That's a great question. I wasn't sure if Firedancer was sensitive to those changes -- @topointon-jump, what do you think?

Comment (topointon-jump) [reply .mergify.yml:194]: These seem to be mostly data structures, helper methods and tests. We are sensitive to the first two for data structures used in the native programs.

File: .mergify.yml [hunk 2]
```diff
@@ -183,6 +183,24 @@ pull_request_rules:
...
+        - files~=^programs/system/src/.*\.rs$
+        - files~=^programs/vote/src/.*\.rs$
+        - files~=^programs/zk-elgamal-proof/src/.*\.rs$
+    actions:
+      comment:
+        message: |
+          If this PR represents a change to a native program implementation, even
+          a refactor, please include someone from the Firedancer team as one of the
+          reviewers.
```
Comment (topointon-jump) [inline .mergify.yml]: Could we change this to encourage developers to keep changes to the native code to a minimum? The wording in the PR description was pretty good. We should also make it clear that this request is limited to just implementation, not tests.

**Review by yihau — DISMISSED** [review]
we can patch it later or update it in this same PR. both work for me!

File: .mergify.yml (hunk 2 above)
Comment (joncinque) [reply .mergify.yml]: Updated the language in [1e2d298](https://github.com/anza-xyz/agave/pull/2233/commits/1e2d29819bfdcfb2f5e895013b9e961c7c11ab98)

File: .mergify.yml (hunk 1 above)
Comment (joncinque) [reply .mergify.yml:194]: Added these to [ce8451f](https://github.com/anza-xyz/agave/pull/2233/commits/ce8451f2f24e2c0debe9bb3c43c34ed5310d9d96)

**Comment by joncinque** [comment]
Sorry for the delay on implementing feedback, but let me know how this looks!

**Review by topointon-jump — APPROVED** [review]

**Review by yihau — APPROVED** [review]

**Pull request merged**